test/functional/rpc_packages.py

```diff
@@ -306,5 +306,80 @@ def test_rbf(self):
+                assert_equal(submitres_tx["vsize"], testres_tx["vsize"])
+                assert_equal(submitres_tx["fees"]["base"], testres_tx["fees"]["base"])
+            entry_info = node.getmempoolentry(submitres_tx["txid"])
+            assert_equal(submitres_tx["vsize"], entry_info["vsize"])
+            assert_equal(submitres_tx["fees"]["base"], entry_info["fees"]["base"])
+
+    def test_submit_child_with_parents(self, num_parents):
+        node = self.nodes[0]
+        # Test a package with num_parents parents and 1 child transaction.
+        package_hex = []
+        package_txns = []
+        values = []
+        scripts = []
+        for _ in range(num_parents):
+            parent_coin = self.coins.pop()
+            value = parent_coin["amount"]
+            (tx, txhex, value, spk) = make_chain(node, self.address, self.privkeys, parent_coin["txid"], value)
+            package_hex.append(txhex)
+            package_txns.append(tx)
+            values.append(value)
+            scripts.append(spk)
+        child_hex = create_child_with_parents(node, self.address, self.privkeys, package_txns, values, scripts)
+        package_hex.append(child_hex)
+        package_txns.append(tx_from_hex(child_hex))
+
+        testres_package = node.testmempoolaccept(rawtxs=package_hex)
+        submitres_package = node.submitpackage(package=package_hex)
+
+        # Check that each result is present, with the correct size and fees
+        for i in range(num_parents + 1):
+            tx = package_txns[i]
+            wtxid = tx.getwtxid()
+            assert wtxid in submitres_package["tx-results"]
+            tx_result = submitres_package["tx-results"][wtxid]
+            expected_fee = Decimal("0.0001")
+            assert_equal(tx_result, {
+                "txid": tx.rehash(),
+                "vsize": tx.get_vsize(),
+                "fees": {
+                    "base": expected_fee,
+                }
+            })
+
+        # submitpackage result should be consistent with testmempoolaccept and getmempoolentry
+        self.assert_equal_package_results(node, testres_package, submitres_package)
+
+        # With no deduplications, the package feerate is just the ancestor feerate of the child.
+        child_txid = testres_package[-1]["txid"]
+        if "package-feerate" in submitres_package:
+            assert_equal(submitres_package["package-feerate"], node.getmempoolentry(child_txid)["fees"]["ancestor"])
+
+    def test_submitpackage(self):
+        node = self.nodes[0]
+
+        self.log.info("Submitrawpackage valid packages with 1 child and some number of parents")
+        for num_parents in [1, 2, 10, 24]:
+            self.test_submit_child_with_parents(num_parents)
+
+        self.log.info("Submitrawpackage only allows packages of 1 child with its parents")
+        # Chain of 3 transactions has too many generations
+        (chain_hex, _) = create_raw_chain(node, self.coins.pop(), self.address, self.privkeys, 3)
```

Comment:
No need for this to be a tuple:

```suggestion
        chain_hex, _ = create_raw_chain(node, self.coins.pop(), self.address, self.privkeys, 3)
```

Reply:
Done